src/wakelock.c

```diff
@@ -61,7 +62,7 @@ void belle_sip_wake_lock_uninit(JNIEnv *env) {
  * @param data Unused.
  */
 static void jni_key_cleanup(void *data) {
-	JNIEnv *env = pthread_getspecific(ctx.jniEnvKey);
+	JNIEnv *env = (JNIEnv*) data;
```

Comment:
Hi Carmen,
The documentation for the pthread_key_create on Android (https://android.googlesource.com/platform/external/pthreads/+/master/pthread_key_create.c) in a bit different:

> This function creates a thread-specific data key visible to all threads. 
All existing and new threads have a value NULL for key until set using pthread_setspecific. 
When any thread with a non-NULL value for key terminates, 'destructor' is called with key's current value for that thread.